train_gpt2.cu

```diff
@@ -2147,6 +2190,11 @@ void gpt2_backward(GPT2 *model) {
     floatX* dresidual = (floatX*)grads_acts.residual3; // the main buffer holding the gradient in the backward pass
     layernorm_backward(dresidual, grads.lnfw, grads.lnfb, scratchF, grads_acts.bt4c, residual, params.lnfw, acts.lnf_mean, acts.lnf_rstd, B, T, C);
 
+    // Aggregate grads.lnfw and grads.lnfb in a background stream
+    floatX* layernorm_backward_pointers[] = {grads.lnfw, grads.lnfb};
+    size_t layernorm_backward_sizes[] = {C, C};
+    multi_gpu_async_all_reduce_pointers_group(2, layernorm_backward_pointers, layernorm_backward_sizes, multi_gpu_config, main_stream);
```

Comment:
comment  says background stream, but  call uses main_stream?

Reply:
oh wait, in the version this code was  based on, main_stream was the background stream?